### PR TITLE
Run Transaction Service IT as part of Ledger API Test Tool

### LINKED ISSUE #1372: Ledger API Test Tool: add TransactionServiceIT into the tool
PoC is at: https://github.com/digital-asset/daml/pull/1019

### PR DESCRIPTION
**Pull request by gleber-da**
Fixes #1372, #1472

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by gleber-da** [comment]
FYI @gerolf-da @gaboraranyossy-da

File: ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala [hunk 1]
```diff
@@ -65,19 +68,29 @@ class TransactionServiceIT
...
 
-  override protected val config: Config =
+  override protected def config: Config =
     Config.default.withTimeProvider(TimeProviderType.WallClock)
 
+  protected val helpers = new TransactionServiceHelpers(config)
+  protected val testTemplateIds = new TestTemplateIds(config)
+  protected val templateIds = testTemplateIds.templateIds
+
+  val runSuffix = "-" + Random.alphanumeric.take(10).mkString
```
Comment (gaboraranyossy-da) [inline ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/TransactionServiceIT.scala]: these utilities code go into a test helper library as we will need them in other places too I suppose.

File: ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala [hunk 2]
```diff
@@ -169,15 +178,39 @@ object LedgerContext {
...
+    def ledgerId: domain.LedgerId =
       configuredLedgerId match {
-        case LedgerIdMode.Static(id) => id.unwrap
+        case LedgerIdMode.Static(id) => id
         case LedgerIdMode.Dynamic() =>
-          LedgerIdentityServiceGrpc
-            .blockingStub(channel)
-            .getLedgerIdentity(GetLedgerIdentityRequest())
-            .ledgerId
+          cachedLedgerId.synchronized {
```
Comment (gaboraranyossy-da) [inline ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala]: wrong concurrency semantics. I mentioned this already in #1400. Why do you have it in this PR too?

Comment (gleber-da) [reply ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala]: One PR depends on another, so one is the subset of another. I will send them as non-draft PRs one by one.

**Comment by gleber-da** [comment]
@gerolf-da This PR is yours! Current state is:
- it contains parts of #1400 and should be rebased on master after #1400 is merged
- can not be merged into master until #1472 is implemented due to existing CI tests which would start failing otherwise

File: docs/source/daml-integration-kit/index.rst [hunk 3]
```diff
@@ -398,17 +398,17 @@ implementation of the :doc:`Ledger API
 </app-dev/index>`. For example, it will show you if
 there are consistency or conformance problem with your implementation.
 
-Assuming that your Ledger API endpoint is accessible at ``localhost:6864``, you can use the tool in the following manner:
+Assuming that your Ledger API endpoint is accessible at ``localhost:6865``, you can use the tool in the following manner:
```
Comment (gerolf-da) [inline docs/source/daml-integration-kit/index.rst]: @gleber-da: I'm assuming `6864` was a typo?

Comment (gleber-da) [reply docs/source/daml-integration-kit/index.rst]: Indeed!

File: docs/source/support/release-notes.rst [hunk 4]
```diff
@@ -20,6 +20,17 @@ Navigator
 - Fixed a regression where Navigator console was not able to inspect contracts and events.
   See `#1454 <https://github.com/digital-asset/daml/issues/1454>`__.
 
+DAML Integration Kit
+~~~~~~~~~~~~~~~~~~~~
+
+- The :doc:`Ledger API Test Tool </tools/ledger-api-test-tool/index>` now also runs ``TransactionServiceIT`` as part of the conformance tests.
+  This requires to load additional ``.dar`` files into the ledger under test. Please refer to the updated instructions in the documentation.
```
Comment (bethaitman) [inline docs/source/support/release-notes.rst]: I think this is what you're saying, reject if incorrect
```suggestion
  This means you need to load additional ``.dar`` files into the ledger under test. Please refer to the updated instructions in the :doc:`documentation </daml-integration-kit/index>`.
```

Comment (gerolf-da) [reply docs/source/support/release-notes.rst]: Thanks, @bethaitman.

**Pull request merged**